### DISCUSSION TITLE
FSESC & Motor selection

### DISCUSSION THREAD

**Discussion by faidoxyz** [prl-mushr/mushr#45] [Hardware]
Hello there! I have few questions to ask/clarify. 

1) Would Flipsky mini [FSESC4.20](https://flipsky.net/collections/electronic-products/products/mini-fsesc4-20-50a-base-on-vesc-widely-used-in-eskateboard-escooter-ebike) would work instead of [FSESC4.12](https://flipsky.net/collections/electronic-products/products/torque-esc-vesc-%C2%AE-bldc-electronic-speed-controller)? 
This [blogpost](https://flipsky.net/blogs/vesc-tool/whats-the-main-difference-between-mini-fsesc4-20-and-fsesc4-12-is-mini-fsesc4-20-reliable) explains the difference between the two FSESC. 

2) I'm planning on using this brushless [motor](https://my.banggood.com/Racerstar-3650-Motor-Brushless-Waterproof-Sensorless-1-or-10-RC-Car-Part-5900-or-4300-or-3900-or-2300KV-p-1110570.html?cur_warehouse=CN&ID=518007&rmmds=search). Would this work? and could you advise which KV I should get? 

Thank you very much

**Comment by palanc** [comment]
Hi @paiyote 

1. We haven't tried using the mini FSESC4.20 ourselves so I can't say whether it would work for sure. My main concern is that it has a hardware version of 4.20, but I am only aware of firmware available for 4.6 - 4.12: [https://github.com/vedderb/bldc-tool/tree/master/firmwares](https://github.com/vedderb/bldc-tool/tree/master/firmwares)
Also note that we need the 'VESC_servoout.bin' binary for whatever hardware you're using because we need to control the servo. Based on the link you provided, it is possible that the same 4.10/4.11/4.12 firmware that we use might work on 4.20, but I can't say for sure. **As VESC-tool will warn you, uploading the incorrect firmware version to a VESC can break it.**

2. Again we haven't used this particular motor before so I can't say for sure. But it seems to be the correct size and I would expect it to work. However, one thing I don't like about it is that (at least as far as I can tell from the pictures) it has a completely round shaft instead of a D-shaped shaft. Having part of the shaft be flat helps more securely mount the motor pinion to the shaft. You could consider a similar model with a D-shaped shaft such as this (though again I have not used this particular motor before): [https://my.banggood.com/Racerstar-3660-Motor-Brushless-Waterproof-Sensorless-18-110-RC-Car-Part-380033002600KV-p-1110647.html?rmmds=detail-topright-recommendation&cur_warehouse=HK&ID=511866](https://my.banggood.com/Racerstar-3660-Motor-Brushless-Waterproof-Sensorless-18-110-RC-Car-Part-380033002600KV-p-1110647.html?rmmds=detail-topright-recommendation&cur_warehouse=HK&ID=511866) 
Any KV rating is probably fine, but you can pick one based off of your intended application. If you need more torque or acceleration, choose a lower KV rating. If you want a higher top speed, choose a higher KV rating.
